File: app/src/main/java/org/bubbble/listlabs/java/CenterSmoothScroller.java
package org.bubbble.listlabs.java;

import android.content.Context;

import androidx.recyclerview.widget.LinearSmoothScroller;

/**
 * @author Andrew
 * @date 2021/02/08 12:13
 */
public class CenterSmoothScroller extends LinearSmoothScroller {

    public CenterSmoothScroller(Context context) {
        super(context);
    }

    @Override
    public int calculateDtToFit(int viewStart, int viewEnd, int boxStart, int boxEnd, int snapPreference) {
        return boxStart + (boxEnd - boxStart) / 2 - (viewStart + (viewEnd - viewStart) / 2);
    }
}

File: app/src/main/java/org/bubbble/listlabs/kotlin/CenterSmoothScroller.kt
package org.bubbble.listlabs.kotlin

import android.content.Context
import androidx.recyclerview.widget.LinearSmoothScroller

/**
 * @author Andrew
 * @date 2021/02/07 17:26
 */
open class CenterSmoothScroller constructor(
    context: Context?
) : LinearSmoothScroller(context) {

    override fun calculateDtToFit(
        viewStart: Int,
        viewEnd: Int,
        boxStart: Int,
        boxEnd: Int,
        snapPreference: Int
    ): Int {
        return boxStart + (boxEnd - boxStart) / 2 - (viewStart + (viewEnd - viewStart) / 2)
    }
}